src/App.tsx

```diff
@@ -1,39 +1,77 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerIdRename = 0;
+
+  componentDidMount() {
+    this.timerIdRename = window.setInterval(this.renameClock, 3300);
+    document.addEventListener('contextmenu', this.handleHideClock);
+    document.addEventListener('click', this.handleShowClock);
+  }
+
+  componentDidUpdate(...args: [{}, State]) {
+    const { clockName: prevName } = args[1];
+    const { clockName: curName } = this.state;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+    if (prevName !== curName) {
+      // eslint-disable-next-line
+      console.log(`Renamed from ${prevName} to ${curName}`);
+    }
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount() {
+    if (this.state.hasClock) {
+      window.clearInterval(this.timerIdRename);
```

Comment:
try to `clearInterval` from the `Clock` component 😉 
your `App` component won't ever be unmounted

Reply:
I don't understand why I have to do this?

Reply:
you don't want clock to be renamed when it is removed 🤔